### PR TITLE
 [MOEB] model: add ViCLIP video-language model (L-14, B-16)

### LINKED ISSUE #5012: Add model: OpenGVLab/ViCLIP-L-14-hf
### Model link on Hugging Face

https://huggingface.co/OpenGVLab/ViCLIP-L-14-hf

### Arxiv link

https://arxiv.org/abs/2307.06942

### Description of the model

ViCLIP is a video-text contrastive model introduced in the InternVid paper (ICLR 2024). It extends CLIP's ViT-L/14 backbone with spatiotemporal attention to encode video clips (8 frames at 224×224) and aligns
  them with text embeddings via contrastive learning on the InternVid-10M-FLT dataset (10M video-text pairs).
  
  **Why add it to MTEB:**
  ViCLIP directly supports T2V and V2T retrieval tasks already in the MOEB benchmark (MSR-VTT, MSVD, VALOR-32K, VSC2022, etc.). It is one of the key video-language models listed in issue #4842 with no existing
  implementation. 
  
  **Models to add:**
  - `OpenGVLab/ViCLIP-L-14-hf` (~400M params, ViT-L/14)
  - `OpenGVLab/ViCLIP-B-16-hf` (~100M params, ViT-B/16)
  
  **License:** CC-BY-NC-SA-4.0
  **Modalities:** video + text
  **Related MOEB issue:** #4842

### PR DESCRIPTION
**Pull request by Rakshitha-Ireddi**
Adds ViCLIP video-language embedding model to MTEB, as part of the MOEB initiative (#4842).
  Closes #5012.
  
  ## Models added
  
  | Model | Params | Embed dim | License |
  |---|---|---|---|
  | `OpenGVLab/ViCLIP-L-14-hf` | 427M | 768 | CC-BY-NC-SA-4.0 |
  | `OpenGVLab/ViCLIP-B-16-hf` | 150M | 512 | CC-BY-NC-SA-4.0 |
  
  ## About ViCLIP
  
  ViCLIP is a video-text contrastive model introduced in the InternVid paper (ICLR 2024, [arXiv:2307.06942](https://arxiv.org/abs/2307.06942)). It extends CLIP's ViT backbone with spatiotemporal attention to
  encode video clips (8 frames at 224×224) and aligns them with text via contrastive learning on InternVid-10M-FLT (10M video-text pairs).
  
  **Tasks covered:** T2VRetrieval, V2TRetrieval — directly applicable to MOEB tasks including MSR-VTT, MSVD, VALOR-32K, VSC2022, and ActivityNet.
  
  ## Implementation notes
  
  ViCLIP uses a non-standard API (custom BPE tokenizer via `model.tokenizer`, `get_vid_features()` / `get_text_features()`) so a custom `ViCLIPWrapper` was implemented following the `xclip_models.py` pattern.
  Requires `trust_remote_code=True`.
  
  ## Checklist
  
  - [x] I have filled out the ModelMeta object to the extent possible
  - [x] I have ensured that my model can be loaded using
    - [x] `mteb.get_model(model_name, revision)` and
    - [x] `mteb.get_model_meta(model_name, revision)`
  - [x] I have tested the implementation works on a representative set of tasks.
  - [x] The model is public — weights available at https://huggingface.co/OpenGVLab/ViCLIP-L-14-hf

### CODE REVIEW

**Comment by Rakshitha-Ireddi** [comment]
Hello @AdnanElAssadi56 and @Samoed . can you please check this

**Comment by Samoed** [comment]
Can you evaluate model on tasks to match performance from papper?

File: mteb/models/model_implementations/viclip_models.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+from mteb.models.model_meta import ModelMeta, ScoringFunction
+
+if TYPE_CHECKING:
+    from torch.utils.data import DataLoader
+
+    from mteb.abstasks.task_metadata import TaskMetadata
+    from mteb.types import Array, BatchedInput, PromptType
+
+_VICLIP_MEAN = (0.48145466, 0.4578275, 0.40821073)
+_VICLIP_STD = (0.26862954, 0.26130258, 0.27577711)
```
Comment (Samoed) [inline mteb/models/model_implementations/viclip_models.py:20]: Where did you take them from?

File: mteb/models/model_implementations/viclip_models.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            for text in text_list:
+                feat = self.model.get_text_features(text, self.tokenizer)
+                if not isinstance(feat, torch.Tensor):
+                    feat = (
+                        feat.pooler_output
+                        if hasattr(feat, "pooler_output")
+                        else next(iter(feat.values()))
+                    )
+                if feat.dim() == 1:
+                    feat = feat.unsqueeze(0)
```
Comment (Samoed) [inline mteb/models/model_implementations/viclip_models.py]: I don't think tat we need this part

File: mteb/models/model_implementations/viclip_models.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            ]
+            # Stack to (B, T, C, H, W) and move to device
+            video_tensor = torch.stack(processed, dim=0).to(self.device)
+            features = self.model.get_vid_features(video_tensor)
+            if not isinstance(features, torch.Tensor):
+                features = (
+                    features.pooler_output
+                    if hasattr(features, "pooler_output")
+                    else next(iter(features.values()))
+                )
```
Comment (Samoed) [inline mteb/models/model_implementations/viclip_models.py]: And this

**Comment by isaac-chung** [comment]
@Rakshitha-Ireddi do you plan on continuing this PR?

**Comment by Rakshitha-Ireddi** [comment]
Hello @isaac-chung , Yes, actively working on the fixes

**Comment by Rakshitha-Ireddi** [comment]
Hi @Samoed, addressed all your review comments in commit c53273fe:
  
  **_VICLIP_MEAN / _VICLIP_STD** - Taken from InternVideo2 preprocessing code, source comment added:
  https://github.com/OpenGVLab/InternVideo/blob/main/InternVideo2/multi_modality/data/process_video.py#L17-L18

  **Defensive tensor checks (lines 98-105 and 129-134)** - Both removed. `get_text_features` and `get_vid_features` always return tensors directly, no defensive cast
  needed.

  **Benchmark evaluation** - Running MSRVTTRetrieval now to verify numbers match the InternVid paper. Will post results here shortly.

**Review by Samoed — APPROVED** [review]

**Comment by Rakshitha-Ireddi** [comment]
@Samoed 
Follow-up: Completed the MSR-VTT evaluation after merge. Results on NVIDIA RTX A5000, 8 frames, zero-shot:
MSRVTTT2V nDCG@10 = 0.562, R@1 = 0.386
MSRVTTV2T nDCG@10 = 0.576, R@1 = 0.412
 
Slightly below the paper's R@1 (0.406 / 0.437) as expected since the paper uses 12 frames and fine-tunes on MSR-VTT train.
